securebanking-openbanking-uk-rs-simulator/src/test/java/com/forgerock/securebanking/openbanking/uk/rs/api/obie/payment/v3_0/domesticpayments/DomesticPaymentsApiControllerTest.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright © 2020 ForgeRock AS (obst@forgerock.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.securebanking.openbanking.uk.rs.api.obie.payment.v3_0.domesticpayments;
+
+import com.forgerock.securebanking.openbanking.uk.rs.persistence.repository.payments.DomesticPaymentSubmissionRepository;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.*;
+import uk.org.openbanking.datamodel.payment.OBWriteDataDomestic1;
+import uk.org.openbanking.datamodel.payment.OBWriteDataDomesticResponse1;
+import uk.org.openbanking.datamodel.payment.OBWriteDomestic1;
+import uk.org.openbanking.datamodel.payment.OBWriteDomesticResponse1;
+
+import java.util.UUID;
+
+import static com.forgerock.securebanking.openbanking.uk.rs.testsupport.api.HttpHeadersTestDataFactory.requiredPaymentHttpHeaders;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+import static uk.org.openbanking.testsupport.payment.OBRisk1TestDataFactory.aValidOBRisk1;
+import static uk.org.openbanking.testsupport.payment.OBWriteDomesticConsentTestDataFactory.aValidOBDomestic1;
+
+/**
+ * A SpringBoot test for the {@link DomesticPaymentsApiController}.
+ */
+@SpringBootTest(webEnvironment = RANDOM_PORT)
```

Comment:
These SpringBootTests use an embedded Mongo and are soooo much quicker than the old docker-compose ones!